lgc/patch/NggPrimShader.cpp

```diff
@@ -4055,7 +4055,14 @@ Function *NggPrimShader::createBackfaceCuller(Module *module) {
   auto func = Function::Create(funcTy, GlobalValue::InternalLinkage, lgcName::NggCullingBackface, module);
 
   func->setCallingConv(CallingConv::C);
+#if LLVM_MAIN_REVISION && LLVM_MAIN_REVISION < 440909
+  // Old version of the code
   func->addFnAttr(Attribute::ReadNone);
+#else
+  // New version of the code (also handles unknown version, which we treat as
+  // latest)
+  func->setDoesNotAccessMemory();
+#endif
```

Comment:
Haven't `setDoesNotAccessMemory()` and friends been around for a while? I suspect we could just use that unconditionally here and elsewhere.